ironic/resource_ironic_node_v1.go

```diff
@@ -67,6 +67,10 @@ func resourceNodeV1() *schema.Resource {
 				Type:     schema.TypeMap,
 				Optional: true,
```

Comment:
You will probably need to surpress updates on root_device inside of properties.  You can specify a `DiffSuppressFunc` here.

Something like:

```golang
DiffSuppressFunc: func(k, _, _ string, _ *schema.ResourceData) bool {
    return k == "properties.root_device"
},
```

Reply:
Ack makes sense, although I'm not sure that's a constraint imposed by Ironic we probably don't want it to change in the lifetime of the resource

Reply:
Oh, it could be changed.  The point here was that this field is managed by the top-level root_device field.  I didn't want terraform to consider it inside properties as well.  

I think the better option is to delete the `root_device` field from the properties map when reading (see below), and only show the value in the top level. Does that make sense?